bundles/org.eclipse.emfcloud.modelserver.emf/src/org/eclipse/emfcloud/modelserver/emf/common/ModelController.java

```diff
@@ -184,37 +182,31 @@ public void saveAll(final Context ctx) {
    }
 
    public void undo(final Context ctx, final String modeluri) {
-      Command undoCommand = modelRepository.undo(modeluri);
+      CCommand undoCommand = modelRepository.getUndoCommand(modeluri);
       if (undoCommand != null) {
-         CCommand encodedCommand;
-         try {
-            encodedCommand = new DefaultCommandCodec().encode(undoCommand);
+         Map<String, JsonNode> encodings = sessionController.getCommandEncodings(modeluri, undoCommand);
```

Comment:
Please add some comment as to why we do the encoding in all formats here. As far as I remember the reasoning was something like this: In order to encode certain commands (e.g., commands that delete something) we need all element information, e.g., the info which element was deleted before it is gone. Therefore we encode the command in all formats and provide them to the model repository which will select the correct format for each subscribed client. This means that we encode the command once for all formats. The alternative would be to encode it once per subscribed client.